steering_controllers/ackermann_steering_controller/CMakeLists.txt

```
@@ -0,0 +1,128 @@
+cmake_minimum_required(VERSION 3.8)
+project(ackermann_steering_controller)
+
+if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  add_compile_options(-Wall -Wextra -Wpedantic)
+endif()
+
+# find dependencies
+set(THIS_PACKAGE_INCLUDE_DEPENDS
+  control_msgs
+  controller_interface
+  geometry_msgs
+  hardware_interface
+  nav_msgs
+  pluginlib
+  rclcpp
+  rclcpp_lifecycle
+  realtime_tools
+  std_srvs
+  tf2
+  tf2_msgs
+  tf2_geometry_msgs
+  ackermann_msgs
+  custom_messages
+  steering_odometry
+  steering_controllers
+)
+
+find_package(ament_cmake REQUIRED)
+find_package(generate_parameter_library REQUIRED)
+foreach(Dependency IN ITEMS ${THIS_PACKAGE_INCLUDE_DEPENDS})
+  find_package(${Dependency} REQUIRED)
+endforeach()
+
+set(msg_files
+  msg/AckermanControllerState.msg
+)
+
+# Add ackermann_steering_controller library related compile commands
+generate_parameter_library(ackermann_steering_controller_parameters
+  src/ackermann_steering_controller.yaml
+  # include/ackermann_steering_controller/validate_ackermann_steering_controller_parameters.hpp
+)
+
+
+
+add_library(
+  ${PROJECT_NAME}
+  SHARED
+  src/ackermann_steering_controller.cpp
+)
+target_include_directories(${PROJECT_NAME} PRIVATE include)
```

```suggestion
target_include_directories(${PROJECT_NAME} PUBLIC
  "$<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>"
  "$<INSTALL_INTERFACE:include/${PROJECT_NAME}>")
```

Update to overriding structure, see the following references:
- https://colcon.readthedocs.io/en/released/user/overriding-packages.html#how-to-make-it-easier-for-your-users-to-override
- https://github.com/PickNikRobotics/generate_parameter_library/commit/1571931483c081a6f50d8cc3aef1dbedb9caddbe